src/main/webapp/script.js

```diff
@@ -1,63 +1,78 @@
-let zoomedIn = false;
-let aqLayerOn = true;
-let routeVisible = false;
-
-function changeBackgroundImage(imagePath) {
-    document.body.style.backgroundImage = `url('${imagePath}')`;
+let map;
+let aqLayer;
+
+function aqLayerControl(controlDiv) {
+    // Set CSS for the control border.
+    const controlUI = document.createElement("div");
+    controlUI.style.backgroundColor = "#fff";
+    controlUI.style.border = "2px solid #fff";
+    controlUI.style.borderRadius = "3px";
+    controlUI.style.boxShadow = "0 2px 6px rgba(0,0,0,.3)";
+    controlUI.style.cursor = "pointer";
+    controlUI.style.margin = "10px";
+    controlUI.style.textAlign = "center";
+    controlUI.title = "Click to toggle the AQ layer";
+    controlDiv.appendChild(controlUI);
+      // Set CSS for the control interior.
+    const controlText = document.createElement("div");
+    controlText.style.color = "rgb(25,25,25)";
+    controlText.style.fontFamily = "Roboto,Arial,sans-serif";
+    controlText.style.fontSize = "16px";
+    controlText.style.lineHeight = "38px";
+    controlText.style.paddingLeft = "5px";
+    controlText.style.paddingRight = "5px";
+    controlText.innerHTML = "Toggle AQ Layer";
+    controlUI.appendChild(controlText);
+
+    controlUI.addEventListener("click", () => {
+        toggleAQLayer();
+    });
 }
 
-function getImagePath() {
-    let path = "";
-    if (zoomedIn) {
-        path = "/images/wagga_";
-    } else {
-        path = "/images/nsw_";
-    }
-
-    if (aqLayerOn) {
-        path += "gradient";
-    } else {
-        path += "plain";
-    }
-
-    if (routeVisible) {
-        path += "_route.png";
-    } else {
-        path += ".png";
-    }
-
-    return path;
+function initMap() {
+    map = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: 37.775, lng: -122.434 },
+        zoom: 13,
+        // zoom for nsw view
+        // center: { lat: -34.397, lng: 150.644 },
+        //zoom: 8,
+        mapTypeControl: true,
+        mapTypeControlOptions: {
+        position: google.maps.ControlPosition.TOP_RIGHT,
+        },
+        streetViewControl: false,
+        fullscreenControl: false,
+    });
+
+    const heatmapData = [
+        {location: new google.maps.LatLng(37.782, -122.447), weight: 0.5},
+        new google.maps.LatLng(37.782, -122.445),
+        {location: new google.maps.LatLng(37.782, -122.443), weight: 2},
+        {location: new google.maps.LatLng(37.782, -122.441), weight: 3},
+        {location: new google.maps.LatLng(37.782, -122.439), weight: 2},
+        new google.maps.LatLng(37.782, -122.437),
+        {location: new google.maps.LatLng(37.782, -122.435), weight: 0.5},
+
```

Comment:
what is this new line for?

Reply:
Good question, didn't notice it before, gone now